src/fragments/lib/graphqlapi/js/optimistic-ui.mdx

```diff
@@ -0,0 +1,539 @@
+        {isLoading && <div>{"Loading Real Estate Property..."}</div>}
+        {isSuccess && realEstateProperty && (
+          <div>
+            <h2>{realEstateProperty.name}</h2>
+            <h3>{realEstateProperty.address}</h3>
+          </div>
+        )}
+        {realEstateProperty && (
+          <div>
+            <button
+              key={`update-${realEstateProperty.id}`}
+              onClick={() =>
+                updateMutation.mutate({
+                  id: realEstateProperty.id,
+                  name: `Updated Home ${Date.now()}`,
+                })
+              }
+            >
+              Update
+            </button>
+            <button
+              key={`delete-${realEstateProperty.id}`}
+              onClick={() =>
+                deleteMutation.mutate({
+                  id: realEstateProperty.id,
+                })
+              }
+            >
+              Delete
+            </button>
+          </div>
+        )}
+      </div>
+    );
+  }
+
+  // NOTE: For test / sample cleanup purposes only (not for docs example)
+  async function deleteAll() {
+    //region: delete realEstateProperties:
+    const response = await API.graphql<
+      GraphQLQuery<ListRealEstatePropertiesQuery>
+    >({
+      query: queries.listRealEstateProperties,
+    });
+
+    console.log(
+      "RealEstateProperties to delete",
+      response?.data?.listRealEstateProperties?.items
+    );
+
+    await response?.data?.listRealEstateProperties?.items.forEach(
+      async (realEstateProperty) => {
+        if (!realEstateProperty?.id) return;
+
+        const realEstatePropertyDetails: DeleteRealEstatePropertyInput = {
+          id: realEstateProperty?.id,
+        };
+
+        const deletedRealEstateProperty = await API.graphql<
+          GraphQLQuery<DeleteRealEstatePropertyMutation>
+        >({
+          query: mutations.deleteRealEstateProperty,
+          variables: { input: realEstatePropertyDetails },
+        });
+
+        console.log("RealEstateProperty deleted: ", deletedRealEstateProperty);
+      }
+    );
+    //endregion
+
+    //region verify all deletes were successful:
+    const secondResponse = await API.graphql<
+      GraphQLQuery<ListRealEstatePropertiesQuery>
+    >({
+      query: queries.listRealEstateProperties,
+    });
+    console.log(
+      "RealEstateProperties should be empty:",
+      secondResponse?.data?.listRealEstateProperties?.items
+    );
+
+    if (secondResponse?.data?.listRealEstateProperties?.items?.length === 0) {
+      console.log("All deletes successful!");
+    }
+    //endregion
+  }
+
+  return (
+    <div className="App">
+      <header className="App-header">
+        {isErrorQuery && <div>{"Problem loading Real Estate Properties"}</div>}
+        {isLoading && <div>{"Loading Real Estate Properties..."}</div>}
+        <ul>
+          {/* TODO: make this work with isSuccess */}
+          {isSuccess &&
```

Comment:
Nit: maybe do an early return for the entire App component if there are errors are isn't succesful.

Reply:
This is not for error handling, but rather for the TS example: https://tanstack.com/query/v4/docs/react/typescript#type-narrowing